In [1]:
# Step 1: Importing libraries and dataset
from sklearn.datasets import load_digits
from sklearn.ensemble import RandomForestClassifier
from sklearn import metrics
import numpy as np
digits  =  load_digits()
train_size = 1500
train_x, train_y = digits.data[:train_size], digits.target[:train_size]
test_x, test_y = digits.data[train_size:], digits.target[train_size:]
np.random.seed(123456)

In [2]:
# Step 2: create the ensemble
ensemble_size = 500
ensemble = RandomForestClassifier(n_estimators=ensemble_size, n_jobs=4)

In [3]:
# Step 3: train the ensemble
ensemble.fit(train_x,train_y)

RandomForestClassifier(n_estimators=500, n_jobs=4)

In [4]:
# Step 4: Evaluate the ensemble
ensemble_predictions = ensemble.predict(test_x)
ensemble_acc = metrics.accuracy_score(test_y, ensemble_predictions)

In [5]:
# Step 5: Print the accuracy
print('Random Forest: %.2f' % ensemble_acc)

Random Forest: 0.93


The classifier achieve 93% which is higher than previous best performing method (XGBoost)